stdlib/LinearAlgebra/src/tridiag.jl

```diff
@@ -591,6 +591,8 @@ similar(M::Tridiagonal, ::Type{T}, dims::Union{Dims{1},Dims{2}}) where {T} = sim
 # Operations on Tridiagonal matrices
 copyto!(dest::Tridiagonal, src::Tridiagonal) = (copyto!(dest.dl, src.dl); copyto!(dest.d, src.d); copyto!(dest.du, src.du); dest)
 
+Base.dataids(A::Tridiagonal) = (Base.dataids(A.dl), Base.dataids(A.d), Base.dataids(A.du))
```

Comment:
Is the idea here that any non-base package also needs to define a similar `dataids` method to avoid an error with `undef` fields?

Reply:
Yes, they would need to (or otherwise not have the undef fields).

Reply:
Maybe we should mention that in the docs along with the other reasons to define custom `dataids`

Reply:
I mean its obscure and no one should do it, but its technically a breaking change